pkg/server/server.go

```diff
@@ -797,8 +818,14 @@ func (s *ProxyServer) serveRecvBackend(backend Backend, stream agent.AgentServic
 				} else {
 					klog.V(5).InfoS("DIAL_CLS sent to frontend", "dialID", resp.Random)
 				}
-				s.PendingDial.Remove(resp.Random)
-				klog.V(5).InfoS("Close streaming", "agentID", agentID, "dialID", resp.Random)
+				if s.PendingDial.Remove(resp.Random) != nil {
+					klog.ErrorS(nil, "Dial terminated (DIAL_CLS) by backend",
+						"dialID", resp.Random,
+						"agentID", agentID,
+						"dialAddress", frontend.dialAddress,
+						"dialDuration", time.Since(frontend.start),
```

Comment:
Same comment  here about whether we should include serverID just for consistency